parquet-protobuf/src/test/java/org/apache/parquet/proto/TestUtils.java

```diff
@@ -145,22 +146,37 @@ private static void checkSameBuilderInstance(MessageOrBuilder[] messages) {
    * Reads messages from given file. The file could/should be created by method writeMessages
    */
   public static <T extends MessageOrBuilder> List<T> readMessages(Path file) throws IOException {
-    ProtoParquetReader<T> reader = new ProtoParquetReader<T>(file);
-
-    List<T> result = new ArrayList<T>();
-    boolean hasNext = true;
-    while (hasNext) {
-      T item = reader.read();
-      if (item == null) {
-        hasNext = false;
-      } else {
-        assertNotNull(item);
-        // It makes sense to return message but production code wont work with messages
-        result.add((T) asMessage(item).toBuilder());
+    return readMessages(file, null);
+  }
+
+  /**
+   * Read messages from given file into the expected proto class.
+   * @param file
+   * @param messageClass
+   * @param <T>
+   * @return List of protobuf messages for the given type.
+   */
+  public static <T extends MessageOrBuilder> List<T> readMessages(Path file, Class<T> messageClass) throws IOException {
+    ParquetReader.Builder readerBuilder = ProtoParquetReader.builder(file);
+    if (messageClass != null) {
+      readerBuilder.set(ProtoReadSupport.PB_CLASS, messageClass.getName()).build();
+    }
+    ParquetReader reader = readerBuilder.build();
+    try {
```

Comment:
we could use try-with-resource construct here

Reply:
Done.